azurelinuxagent/ga/exthandlers.py

```diff
@@ -510,27 +519,30 @@ def handle_ext_handlers(self, goal_state_id):
         depends_on_err_msg = None
         extensions_enabled = conf.get_extensions_enabled()
 
-        # Instantiate policy engine, and use same engine to handle all extension handlers. If an error is thrown during
-        # policy engine initialization, we block all extensions and report the error via handler status for each extension.
-        # Save policy to history folder.
+        # Instantiate policy engine. If an error is thrown during policy engine initialization, we block all extensions
+        # and report the error via handler status for each extension.
```

Comment:
we should add an extra check below and block the extensions only if the user has opted in to policy